apps/wallet/src/ui/app/components/active-coins-card/CoinItem.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+
+import Loading from '../loading';
+import { Text } from '_app/shared/text';
+import { CoinIcon } from '_components/coin-icon';
+import { useRpc, useFormatCoin } from '_hooks';
+
+// Get coin metadata from the coin type
+export function CoinItem({
+    coinType,
+    balance,
+}: {
+    coinType: string;
+    balance: bigint;
+}) {
+    const rpc = useRpc();
+
+    // Get coin metadata from the coin type
+    // Include icon url and name of the coin
+    const { data, isLoading } = useQuery(
```

Comment:
Ideally this should go to it's own file I think. (We should probably add a folder in hooks and put there all the hooks related to api)